packages/config-array/src/config-array.js

```diff
@@ -88,8 +88,13 @@ const CONFIG_WITH_STATUS_UNCONFIGURED = Object.freeze({
 	status: "unconfigured",
 });
 
+// Match two leading dots followed by a path separator or the end of input.
 const EXTERNAL_PATH_REGEX = new RegExp(`^\\.\\.($|\\${path.sep})`, "u");
 
+// Some `node:path` polyfills do not provide an implementation for `toNamespacedPath`.
+// In that case, return the argument unchanged like on POSIX systems.
+const toNamespacedPath = path.toNamespacedPath ?? (arg => arg);
```

Comment:
I'm confused. If we can just replace this with `arg => arg`, then why should we use `toNamespacedPath` at all?

Reply:
`toNamespacedPath` is only required to handle paths on Windows. Browser polyfills always use Unix-style paths with `/` as a separator and no drive letters, probably because that's how URL paths are written, and in that case it's safe to compare paths without changes. The assumption here is that when `toNamespacedPath` is not defined, we are dealing with Unix paths.

Reply:
Hmm, I'm not sure this is a safe assumption. This package is intended to be used in browsers, in which case we won't know what format the path string is when passed.

Is it possible to inline the `toNamespacedPath` function? Or an approximation?

Reply:
Thanks! I've added a functionally equivalent polyfill for Windows based on [Node.js implementation](https://github.com/nodejs/node/blob/v22.3.0/lib/path.js#L622-L648). Once `toNamespacedPath` is better supported by polyfill libraries, we can simply delete that and switch to using the `node:path` export.